tests/transforms/test_qcut.py

```diff
@@ -1297,8 +1311,9 @@ class TestContractTensors:
     """Tests for the contract_tensors function"""
 
     t = [np.arange(4), np.arange(4, 8)]
-    m = [[qcut.MeasureNode(wires=0)], []]
-    p = [[], [qcut.PrepareNode(wires=0)]]
+    # make copies of nodes to ensure id comparisons work correctly
+    m = [[copy.copy(qcut.MeasureNode(wires=0))], []]
+    p = [[], [copy.copy(qcut.PrepareNode(wires=0))]]
     edge_dict = {"pair": (m[0][0], p[1][0])}
```

Comment:
I don't think `copy.copy(some_class())` is practically any different to `some_class()` :thinking:
We should define `m` and `p` without copies, and use a copy in the `edge_dict` :+1:

Reply:
fixed here 193631e58922d4941f6f0c515efea8b8833e10c8